output_template/python/run.py

```diff
@@ -89,9 +87,33 @@ def main_test(input_image, library, config_file, max_percent_incorrect_values=0.
     # add the batch dimension
     data = np.expand_dims(data, axis=0)
 
+    if file_extension == '.so':  # Shared library
+        # load and initialize the generated shared model
+        nn = NNLib()
+        nn.load(model)
+        nn.init()
+
+    elif file_extension == '.pb':  # Protocol Buffer file
+        # only load tensorflow if user wants to use GPU
+        from lmnet.tf_graph_load_pb import TFGraphLoadPb
+        nn = TFGraphLoadPb(model)
+
     # run the graph
     output = nn.run(data)
 
+    return output, raw_image
+
+
+def main_test(input_image, model, config_file, max_percent_incorrect_values=0.1):
```

Comment:
`main_test()` is bad method name ... 😭 
Could you change the name ? When you have the time

Reply:
Got it. I'll change it to `run_prediction`. is it okay?

Reply:
Thanks 👍